Tests/tests/test_TraceControl.cpp

```diff
@@ -0,0 +1,327 @@
+ * file the following copyright and licenses apply:
+ *
+ * Copyright 2022 RDK Management
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+#include "gtest/gtest.h"
+#include <gmock/gmock.h>
+
+#include "TraceControl.h"
+
+#include "FactoriesImplementation.h"
+#include "ServiceMock.h"
+#include "COMLinkMock.h"
+
+using namespace WPEFramework;
+
+namespace {
+const string callSign = _T("TraceControl");
+const string webPrefix = _T("/Service/TraceControl");
+const string volatilePath = _T("/tmp/");
+}
+
+class TraceControlTest : public ::testing::Test {
+protected:
+    Core::ProxyType<Plugin::TraceControl> plugin;
+    PluginHost::IWeb* interface;
+    Core::JSONRPC::Handler& handler;
+    Core::JSONRPC::Connection connection;
+    string response;
+
+    TraceControlTest()
+        : plugin(Core::ProxyType<Plugin::TraceControl>::Create())
+        , handler(*(plugin))
+        , connection(1, 0)
+    {
+        interface = static_cast<PluginHost::IWeb*>(plugin->QueryInterface(PluginHost::IWeb::ID));
+        Trace::TraceUnit::Instance().Open(volatilePath);
+    }
+
+    virtual ~TraceControlTest()
+    {
+        interface->Release();
+        plugin.Release();
+        Trace::TraceUnit::Instance().Close();
+    }
+};
+
+class TraceControlInitializedTest : public TraceControlTest {
+protected:
+    FactoriesImplementation factoriesImplementation;
+    ServiceMock service;
+    COMLinkMock comLinkMock;
+
+    TraceControlInitializedTest()
+        : TraceControlTest()
+    {
+        ON_CALL(service, ConfigLine())
+            .WillByDefault(::testing::Return("{}"));
+        ON_CALL(service, WebPrefix())
+            .WillByDefault(::testing::Return(webPrefix));
+        ON_CALL(service, VolatilePath())
+            .WillByDefault(::testing::Return(volatilePath));
+        ON_CALL(service, Callsign())
+            .WillByDefault(::testing::Return(callSign));
+         ON_CALL(service, COMLink())
+             .WillByDefault(::testing::Return(&comLinkMock));
+
+        PluginHost::IFactories::Assign(&factoriesImplementation);
+    }
+    virtual ~TraceControlInitializedTest() override
+    {
+        plugin->Deinitialize(&service);
+        PluginHost::IFactories::Assign(nullptr);
+    }
+};
+
+TEST_F(TraceControlInitializedTest, registeredMethods)
+{
+    ON_CALL(service, Background())
+        .WillByDefault(::testing::Return(true));
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
```

Comment:
you don't need TraceControlInitializedTest, ON_CALL, and Initialize to test handler.Exists. please remove

Reply:
I'll do it